### PR TITLE
Basic naive, single-threaded queue implementation. Slow.

### PR DESCRIPTION
**Pull request by maxdjohnson**
Added very basic queue logic. It's not parallel, nor threadsafe. It's abysmally slow, but it works.

### CODE REVIEW

File: gc.c [hunk 1]
```diff
@@ -1650,6 +1670,45 @@ struct mark_tbl_arg {
 }
 
 void
+gc_mark_defer(rb_objspace_t *objspace, VALUE ptr, int lev) {
+    mark_queue_node_t* node =
+	(mark_queue_node_t*) malloc(sizeof(mark_queue_node_t));
+    node->objspace = objspace;
```
Comment (kevints) [inline gc.c:1676]: Lots of extra copies of objspace, potentially refactor later to use a st_table <objspace*, mark_queue_node_t*> and remove that pointer from mark_queue_node_t to save space?

**Comment by kevints** [comment]
LGTM, go ahead and merge.

File: gc.c [hunk 2]
```diff
@@ -1650,6 +1670,45 @@ struct mark_tbl_arg {
 }
 
 void
+gc_mark_defer(rb_objspace_t *objspace, VALUE ptr, int lev) {
+    mark_queue_node_t* node =
+	(mark_queue_node_t*) malloc(sizeof(mark_queue_node_t));
```
Comment (kevints) [inline gc.c:1675]: We need to think about refactoring this - a tiny malloc on every node overshadows any possible gains from parallelism

Comment (andrewmains12) [inline gc.c:1675]: I agree--I think the best approach is probably to use a circular buffer to implement 
fixed size queues (similar to what the current implementation does for the mark stack.)

**Pull request merged**